msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java

```diff
@@ -95,4 +100,114 @@ public void onError(Exception exception) {
                     }
                 });
     }
+
+    private MsalOAuth2TokenCache getTokenCache(@NonNull final Context appContext,
+                                               @NonNull final IAccountCredentialCache accountCredentialCache) {
+        return new MsalOAuth2TokenCache(
+                appContext,
+                accountCredentialCache,
+                new MicrosoftStsAccountCredentialAdapter());
+    }
+
+    private IAccountCredentialCache getCurrentAccountCredentialCache(final Context appContext){
+        return new SharedPreferencesAccountCredentialCache(
+                new CacheKeyValueDelegate(),
+                new SharedPreferencesFileManager(
+                        appContext,
+                        SINGLE_ACCOUNT_CREDENTIAL_SHARED_PREFERENCES,
+                        new StorageHelper(appContext)
+                )
+        );
+    }
+
+    /**
+     * Get current account that is persisted in shared preference.
+     *
+     * @param appContext
+     * @return a persisted MultiTenantAccount. This could be null.
+     * */
+    @Nullable
+    private MultiTenantAccount getPersistedCurrentAccount(@NonNull final Context appContext) {
+        final String methodName = ":getPersistedCurrentAccount";
+        final IAccountCredentialCache accountCredentialCache = getCurrentAccountCredentialCache(appContext);
+        final MsalOAuth2TokenCache msalOAuth2TokenCache = getTokenCache(appContext, accountCredentialCache);
+
+        List<AccountRecord> accountRecords = accountCredentialCache.getAccounts();
+        if (accountRecords.size() == 0) {
+            return null;
+        } else if (accountRecords.size() > 1) {
```

Comment:
🐛 Not true -- even in `SingleAccountMode` a single _user principle_ may have multiple `AccountRecords`.

See [this comment](https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/651/files#r292682883) below.

Let's say a user signs in to Teams (in `SingleAccountMode`/`SharedDeviceMode`) and that user is in `Contoso` tenant, if that user is a _guest_ in `Fabrikam`, they may authorize to that tenant as well, in which case, this user will have two `AccountRecords`, 2 `IdTokenRecords`, 2 `AccessTokenRecords` and 1 `RefreshTokenRecord`.

Use case here might a user who is FLW, but needs to chat with employees in multiple tenants at once

Reply:
It's worth calling out that _all_ of a user's accounts will have the same `home_account_id`, irrespective of tenant. The `local_account_id` will change

Reply:
For additional clarity on this point, see [`MsalOAuth2TokenCache#getAccountsWithAggregatedAccountData()`](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/iambmelt/tenant-profile/common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java#L621-L651) -- this method loads all of the `AccountRecords` & `IdTokenRecords` for a principal so as to later construct the full `IAccount` inside of [`AccountAdapter#adapt()`](https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/iambmelt/tenant-profile-redux/msal/src/main/java/com/microsoft/identity/client/AccountAdapter.java#L143-L183)